### DISCUSSION TITLE
BaseEntity.toObject() vs BaseEntity.toPOJO()

### DISCUSSION THREAD

**Discussion by arinanto** [mikro-orm/mikro-orm#4604] [Q&A] [answered]
I'm having difficulty finding references on the difference between `toObject()` and `toPOJO()`. I've read `BaseEntity.ts` code and found that `toJSON()` by default is just calling to `toObject()`. Yet `toPOJO()` have its own distinct call, how does this two differ in behaviour?

**Comment by B4nan** [comment, marked as answer]
`toObject()` will respect things like partial loading and populate hints or hidden properties, unlike the `toPOJO`. The `toPOJO` method was added for one use case - caching. When you load some entity, let's say a `User` that has a `password` which is `hidden: true`, calling `toObject()` would omit this hidden property, while `toPOJO` would keep it. If you want to cache such entity, you want to have all the properties, not just those that are visible.

They both use `EntityTransformer.toObject()`, the difference is only in the `raw` parameter.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/WrappedEntity.ts#L72
https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/WrappedEntity.ts#L76
https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/serialization/EntityTransformer.ts

**Reply by arinanto** [reply]
Thanks for the swift response. Yeah, I guess that make total sense. So we can separate between internal and external representation depending on the use case. I actually comes across multiple cases for this, so knowing this is quite helpful.
